renderdoc/driver/metal/metal_device.cpp

```diff
@@ -179,7 +179,13 @@ bool WrappedMTLDevice::Serialise_newCommandQueue(SerialiserType &ser, WrappedMTL
 
   if(IsReplayingAndReading())
   {
-    // TODO: implement RD MTL replay
+    MTL::CommandQueue *realMTLCommandQueue = Unwrap(this)->newCommandQueue();
```

Comment:
A question for all the code which is creating the live resources.
What code should be written or asserts added to handle or guard if the live resource is created multiple times? 
Currently, this resource creation would leak a live resource if it is called multiple times with the same resource ID.

Reply:
That should not be possible, since IDs are monotonically increasing (can't be re-used) and creation chunks should only be created once. That would indicate some bug at capture time that caused a duplicate chunk or something like that.

The resource manager already has an error message if a live ID is registered twice, which would be enough to not have this lost silently.